### PR TITLE
go/athenzutils: add ExtractHostname function

### PR DESCRIPTION
**Pull request by dmitris**
PR addresses https://github.com/AthenZ/athenz/issues/1508 - adds the `ExtractHostname` helper function to the Go athenzutils package.  The unit test reuses an existing certificate from `servers/zts/src/test/resources/athenz.examples.uri-hostname-only.pem`.

Test coverage in `libs/go/athenzutils` (per `go test -cover`):
master - 60.5% 
this PR - 62.8%

@havetisyan

### CODE REVIEW

File: libs/go/athenzutils/hostname.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2021 Verizon Media, Inc.
```
Comment (havetisyan) [inline libs/go/athenzutils/hostname.go:1]: Must be: Copyright Athenz Authors

File: libs/go/athenzutils/hostname_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 Verizon Media, Inc.
```
Comment (havetisyan) [inline libs/go/athenzutils/hostname_test.go:1]: Must be: Copyright Athenz Authors

File: libs/go/athenzutils/hostname_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	tests := []struct {
+		name     string
+		certFile string
+		hostname string
+	}{
+		{"nocn", "data/no_cn_x509.cert", ""},
+		{"emailnouri", "data/valid_email_x509.cert", ""},
+		{
+			"urihostname",
+			"../../../servers/zts/src/test/resources/athenz.examples.uri-hostname-only.pem",
```
Comment (havetisyan) [inline libs/go/athenzutils/hostname_test.go:19]: let's just copy this cert into data folder to make each module independent. we shouldn't have test cases in one module depend on resources from another module.

**Review by havetisyan — CHANGES_REQUESTED** [review]

File: libs/go/athenzutils/hostname.go (hunk 1 above)
Comment (dmitris) [reply libs/go/athenzutils/hostname.go:1]: without a year? Or `// Copyright 202 1 Athenz Authors`?

File: libs/go/athenzutils/hostname_test.go (hunk 2 above)
Comment (dmitris) [reply libs/go/athenzutils/hostname_test.go]: changed

File: libs/go/athenzutils/hostname_test.go (hunk 3 above)
Comment (dmitris) [reply libs/go/athenzutils/hostname_test.go]: done

**Review by havetisyan — APPROVED** [review]
Thanks

**Pull request merged**